src/Apps/Order/Components/AddressVerificationFlow.tsx

```diff
@@ -479,3 +439,73 @@ export const AddressVerificationFlowQueryRenderer: React.FC<{
     />
   )
 }
+
+const fallbackFromFormValues = (address: AddressValues): AddressOption => {
+  return {
+    key: USER_INPUT,
+    recommended: false,
+    lines: [
+      address.addressLine1,
+      address.addressLine2 || null,
+      `${address.city}, ${address.region} ${address.postalCode}`,
+      address.country,
+    ].filter(Boolean) as string[],
+
+    address: {
+      addressLine1: address.addressLine1,
+      addressLine2: address.addressLine2,
+      region: address.region,
+      country: address.country,
+      city: address.city,
+      postalCode: address.postalCode,
+    },
+  }
+}
+
+const useAddressVerificationTracking = () => {
+  const { trackEvent } = useTracking()
+  const { contextPageOwnerId } = useAnalyticsContext()
+  const { user } = useSystemContext()
+  const userId = user?.id
+
+  return {
+    trackViewedModal: useCallback(
+      ({ subject }: { subject: string }) => {
+        trackEvent({
+          action: ActionType.validationAddressViewed,
+          context_module: ContextModule.ordersShipping,
+          context_page_owner_type: OwnerType.ordersShipping,
+          context_page_owner_id: contextPageOwnerId,
+          user_id: userId,
+          flow: "user adding shipping address",
+          subject,
+        } as ValidationAddressViewed)
+      },
+      [contextPageOwnerId, trackEvent, userId]
+    ),
```

Comment:
tracking logic moved into this local helper to clean up the main component. I'm not sure if all these `useCallback()`s are necessary.